src/replication.c

```diff
@@ -2140,8 +2148,6 @@ int slaveTryPartialResynchronization(connection *conn, int read_reply) {
                 server.master_replid,
                 server.master_initial_offset);
         }
-        /* We are going to full resync, discard the cached master structure. */
-        replicationDiscardCachedMaster();
```

Comment:
if we avoid doing that now, maybe we should avoid doing other things which the caller of this function does right after.
e.g.
```c
    disconnectSlaves(); /* Force our slaves to resync with us as well. */
    freeReplicationBacklog(); /* Don't allow our chained slaves to PSYNC. */
```

Reply:
Yes, for sub-replicas, forcing sub-replica to resync with this replicas should be after finishing transferring RDB and preparing to load RDB.